Initial commit

joehowarth17 opened the pull request.

joehowarth17 committed:
Initial commit

rahmant3 commented on GUI/python source/UMSATSGUI.py:
```diff
@@ -20,45 +20,43 @@
 
```
There should be a file header for this. Containing information such as:
- File History (i.e. date of commit, author name, etc).

Chat with me if you need more info.

rahmant3 replied:
(if these are autogenerated, I'd suggest mentioning that as well)

rahmant3 commented on GUI/python source/UMSATSGUI.py:
```diff
@@ -20,45 +20,43 @@
 def vp_start_gui():
-    '''Starting point when module is the main routine.'''
+    """Starting point when module is the main routine."""
     global val, w, root
```
Need to be very careful about global variables. They should be given descriptive names, and they should be organized together (e.g. after the imports, but before the functions).

rahmant3 commented on GUI/python source/UMSATSGUI.py:
```diff
@@ -20,45 +20,43 @@
+        """This class configures and populates the toplevel window.
+           top is the toplevel containing window."""
+
         _bgcolor = '#d9d9d9'  # X11 color: 'gray85'
```
These colors could be separated into constants, e.g. grouped together at the top of the file, like so:

COLOR_GRAY_85 = "#d9d9d9"
COLOR_BLACK = "#000000"

This will make it easier to adjust the colors later on, without the need to do a find-replace. Goes for other colors in this file.

rahmant3 commented on GUI/python source/UMSATSGUI.py:
```diff
@@ -101,8 +95,7 @@ def __init__(self, top=None):
         self.Listbox2 = Listbox(self.frameTerminal)
-        self.Listbox2.place(relx=0.03, rely=0.59, relheight=0.35, relwidth=0.92
-                , y=-12, h=12)
+        self.Listbox2.place(relx=0.03, rely=0.59, relheight=0.35, relwidth=0.92, y=-12, h=12)
```
I would argue these are magic numbers as well.

rahmant3 commented on GUI/python source/UMSATSGUI.py:
```diff
@@ -164,7 +155,6 @@ def __init__(self, top=None):
-        #self.Command2Label.place(relx=0.03, rely=0.87, height=21, width=124, y=-12)
         self.Command2Label.configure(activebackground="#f9f9f9")
         self.Command2Label.configure(activeforeground="black")
         self.Command2Label.configure(background="#d9d9d9")
```
For example, constants for those colors would be pretty useful here...

rahmant3 commented on GUI/python source/UMSATSGUI.py:
```diff
@@ -702,8 +655,7 @@ def __init__(self, top=None):
         self.Message3.configure(width=64)
 
         # self.Checkbutton1 = Checkbutton(self.LabelStatus)
```
Why is there commented code here? If it isn't needed, it should be removed.

rahmant3 commented on GUI/python source/UMSATSGUI_support.py:
```diff
@@ -52,24 +54,19 @@ def set_Tk_var():
-    sys.stdout.flush()
 
 def init(top, gui, *args, **kwargs):
     global w, top_level, root, running
```
With this many global variables, I think it would be easier if this UMSATS support stuff was in its own class.

rahmant3 commented on GUI/python source/UMSATSGUI_support.py:
```diff
@@ -83,7 +80,12 @@ def init(top, gui, *args, **kwargs):
     logFile2 = None
 
-    top_level.bind_all('<<newSerialData>>', updateDisplay)
+    date = str(datetime.datetime.today().month) + "-" + str(datetime.datetime.today().day) + "_" + str(datetime.datetime.today().hour) + "h" + str(datetime.datetime.today().minute) + "m"
```
I think you can use strftime to format your date string, instead of going to this trouble: https://stackoverflow.com/questions/311627/how-to-print-a-date-in-a-regular-format

rahmant3 commented on GUI/python source/UMSATSGUI_support.py:
```diff
@@ -93,78 +95,88 @@ def init(top, gui, *args, **kwargs):
+            can_id = ""
+
+        # For now ignore any mesasge with ID except Serial_ID.
+        if (can_id == 0x100) or (can_id == 0x200) or (can_id == 0x300) or (can_id == 0x400) or (can_id == 0x500):
```
The magic numbers ( the CAN id's) should be moved into separate constants. Also, perhaps you can place them in a global list (e.g. named "ACCEPTED_IDS") so that you can do "if can_id in ACCEPTED_IDS".

rahmant3 commented on GUI/python source/UMSATSGUI_support.py:
```diff
@@ -174,19 +186,21 @@ def destroy_window():
-    #print('command1_enter\n')
-    writeCommand(e.widget,display,1,cmd1Text.get())
+
+def command1_enter(e, display):
```
These function names should be more descriptive name than "command1" and "command2"?

rahmant3 commented on GUI/python source/UMSATSGUI_support.py:
```diff
@@ -174,19 +186,21 @@ def destroy_window():
+
 
-def writeCommand(inputBox,display,can,data):
+def write_command(input_box, display, can, data):
```
For some of these function headers, can you also explain what each parameter is for? This one in particular isn't clear.

joehowarth17 committed:
Adressed comments from review and updated arduino code.

- Added 'file header' comments.
- Added comments for unclear functions.
- Removed commented and unused code.
- Added constants for CAN IDs and other commands etc.

rahmant3 commented on GUI/Arduino/UMSATS_GUI.ino:
```diff
@@ -244,67 +304,94 @@ void printFrame(CAN_FRAME &frame) {
-  radio();}
-//  while(Can0.available() <= 0){}
+
+  if(comms_sm_input == 'a'{
```
Looks like it's missing a parenthesis. I think it doesn't compile? Before uploading, make sure to compile everything to make sure there aren't any syntax errors. Similarly, you should upload a new binary of the GUI (the *.exe file) as well.

rahmant3 commented on GUI/Arduino/UMSATS_GUI.ino:
```diff
@@ -244,67 +304,94 @@ void printFrame(CAN_FRAME &frame) {
 
-  
-  }
+	switch(rx){
```
Looks like there is a mix of tabs / spaces that's causing the spacing to be weird. This is indented as a tab, whereas the line before is indented as spaces. You can do a "replace all" fairly easily on all the tabs if you open up this file in Notepad++ (i.e. replace '\t' with two or four spaces).

rahmant3 commented on GUI/python source/UMSATSGUI_support.py:
```diff
@@ -130,10 +152,10 @@ def update_display(event):
 
         # For now ignore any mesasge with ID except Serial_ID.
-        if (can_id == 0x100) or (can_id == 0x200) or (can_id == 0x300) or (can_id == 0x400) or (can_id == 0x500):
+        if can_id in ACCEPTED_IDS:
```
Nice, much cleaner way of doing this. Also, typo in the line above.

rahmant3 commented on GUI/Arduino/UMSATS_GUI.ino:
```diff
@@ -200,39 +251,48 @@ void sendGround(int command, int well,int seconds){
+}
   
+void printFrame(CAN_FRAME &frame) {
+  //This function preints out a CAN frame to the GUI.
```
Typo: "prients" -> "prints". I'm not going to point all of them out, just mentioning that in general, you should make sure you use proper English (grammar and spelling), because it keeps the documentation professional and easier to read.

rahmant3 reviewed (changes requested):
Good job. Overall, everything is much cleaner. Just a few minor things to tweak, and then I will merge this.

joehowarth17 committed:
Fixed typos and created new executable.
